[ml][release-test] air_example_gptj_deepspeed_fine_tuning.aws failed

### What happened + What you expected to happen

air_example_gptj_deepspeed_fine_tuning.aws is failing consistently

https://buildkite.com/ray-project/release-tests-branch/builds/1650#0188142d-408c-4df3-9bdc-b63d7b463125

[Bisect](https://buildkite.com/ray-project/release-tests-bisect/builds/154#_) blamed this https://github.com/ray-project/ray/commit/bfd6a63fbbae51f79eb738643361ee3b15f7a146. Please help review and resolve the issue.

### Versions / Dependencies

master

### Reproduction script

https://buildkite.com/ray-project/release-tests-branch/builds/1650#0188142d-408c-4df3-9bdc-b63d7b463125

### Issue Severity

High: It blocks me from completing my task.

```python
2023-05-13 01:52:46,342 ERROR tune_controller.py:859 -- Trial task failed
Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/air/execution/_internal/event_manager.py", line 110, in resolve_future
    result = ray.get(future)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/_private/auto_init_hook.py", line 18, in auto_init_wrapper
    return fn(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/_private/client_mode_hook.py", line 99, in wrapper
    return func(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/_private/worker.py", line 2518, in get
    raise value.as_instanceof_cause()
ray.exceptions.RayTaskError(AssertionError): ray::_Inner.train() (pid=24386, ip=10.0.16.114, actor_id=0f3b6cac57dec24813e64f1402000000, repr=TransformersTrainer)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/tune/trainable/trainable.py", line 388, in train
    raise skipped from exception_cause(skipped)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/train/_internal/utils.py", line 54, in check_for_failure
    ray.get(object_ref)
ray.exceptions.RayTaskError(AssertionError): ray::_RayTrainWorker__execute.get_next() (pid=4815, ip=10.0.49.30, actor_id=862ab78426c5717f0525c77902000000, repr=<ray.train._internal.worker_group.RayTrainWorker object at 0x7fc6e7c3b2b0>)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/train/_internal/worker_group.py", line 32, in __execute
    raise skipped from exception_cause(skipped)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/train/_internal/utils.py", line 129, in discard_return_wrapper
    train_func(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.9/site-packages/ray/train/hf_transformers/transformers_trainer.py", line 390, in _huggingface_train_loop_per_worker
    trainer: transformers.trainer.Trainer = trainer_init_per_worker(
  File "/tmp/tmp4_9lv1ws", line 336, in trainer_init_per_worker
  File "/home/ray/anaconda3/lib/python3.9/site-packages/transformers/modeling_utils.py", line 2606, in from_pretrained
    init_contexts = [deepspeed.zero.Init(config_dict_or_path=deepspeed_config())] + init_contexts
  File "/home/ray/anaconda3/lib/python3.9/site-packages/deepspeed/runtime/zero/partition_parameters.py", line 724, in __init__
    _ds_config = deepspeed.runtime.config.DeepSpeedConfig(config_dict_or_path,
  File "/home/ray/anaconda3/lib/python3.9/site-packages/deepspeed/runtime/config.py", line 764, in __init__
    self._configure_train_batch_size()
  File "/home/ray/anaconda3/lib/python3.9/site-packages/deepspeed/runtime/config.py", line 937, in _configure_train_batch_size
    self._batch_assertion()
  File "/home/ray/anaconda3/lib/python3.9/site-packages/deepspeed/runtime/config.py", line 885, in _batch_assertion
    assert train_batch == micro_batch * grad_acc * self.world_size, (
AssertionError: Check batch related parameters. train_batch_size is not equal to micro_batch_per_gpu * gradient_acc_step * world_size 256 != 16 * 1 * 1
```

It seems that deepspeed doesn't recognize world_size correctly. (should have been 16 but turned out to be 1). cc @Yard1

--- a/doc/source/ray-air/examples/gptj_deepspeed_fine_tuning.ipynb
+++ b/doc/source/ray-air/examples/gptj_deepspeed_fine_tuning.ipynb
@@ -30,6 +30,7 @@
    ]
   },
   {
+   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
@@ -42,7 +43,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "#! pip install \"datasets\" \"evaluate\" \"accelerate>=0.16.0\" \"transformers>=4.26.0\" \"torch>=1.12.0\" \"deepspeed\""
+    "#! pip install \"datasets\" \"evaluate\" \"accelerate==0.18.0\" \"transformers>=4.26.0\" \"torch>=1.12.0\" \"deepspeed==0.8.3\""
    ]
   },
   {
@@ -57,6 +58,7 @@
    ]
   },
   {
+   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
@@ -78,6 +80,7 @@
    ]
   },
   {
+   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
@@ -153,10 +156,14 @@
     "        \"pip\": [\n",
     "            \"datasets\",\n",
     "            \"evaluate\",\n",
-    "            \"accelerate>=0.16.0\",\n",
-    "            \"transformers>=4.26.0\",\n",
+    "            # Latest combination of accelerate==0.19.0 and transformers==4.29.0\n",
+    "            # seems to have issues with DeepSpeed process group initialization,\n",
+    "            # and will result in a batch_size validation problem.\n",
+    "            # TODO(jungong) : get rid of the pins once the issue is fixed.\n",
+    "            \"accelerate==0.16.0\",\n",
+    "            \"transformers==4.26.0\",\n",
     "            \"torch>=1.12.0\",\n",
-    "            \"deepspeed\",\n",
+    "            \"deepspeed==0.9.2\",\n",
     "        ]\n",
     "    }\n",
     ")"
@@ -234,6 +241,7 @@
    ]
   },
   {
+   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
@@ -308,6 +316,7 @@
    ]
   },
   {
+   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
@@ -340,6 +349,7 @@
    ]
   },
   {
+   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
@@ -969,6 +979,7 @@
    ]
   },
   {
+   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
@@ -997,6 +1008,7 @@
    ]
   },
   {
+   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [

--- a/release/air_examples/gptj_deepspeed_finetuning/gptj_deepspeed_env.yaml
+++ b/release/air_examples/gptj_deepspeed_finetuning/gptj_deepspeed_env.yaml
@@ -7,10 +7,10 @@ python:
   pip_packages:
     - "datasets"
     - "evaluate"
-    - "accelerate>=0.16.0"
-    - "transformers>=4.26.0"
+    - "accelerate==0.16.0"
+    - "transformers==4.26.0"
     - "torch>=1.12.0"
-    - "deepspeed"
+    - "deepspeed==0.9.2"
     - myst-parser==0.15.2
     - myst-nb==0.13.1
     - jupytext==1.13.6

